core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/WorkspaceBillingAccountMonitorSpec.scala

```diff
@@ -0,0 +1,177 @@
+
+      val actor = createWorkspaceBillingAccountMonitor(dataSource, failingGcsDAO)
+
+      eventually (timeout = timeout(Span(10, Seconds))) {
+        runAndWait(workspaceQuery.findByName(secondV1Workspace.toWorkspaceName)).getOrElse(fail("workspace not found"))
+          .billingAccountErrorMessage shouldBe Option(failureMessage)
+        runAndWait(workspaceQuery.findByName(v1Workspace.toWorkspaceName)).getOrElse(fail("workspace not found"))
+          .billingAccountErrorMessage shouldBe Option(failureMessage)
+      }
+      verify(failingGcsDAO, times(1)).updateGoogleProjectBillingAccount(secondV1Workspace.googleProjectId, Option(newBillingAccount))
+
+      system.stop(actor)
+    }
+  }
+
+  it should "mark a billing project's billing account as invalid if Google returns a 403" in {
+    withEmptyTestDatabase { dataSource: SlickDataSource =>
+      val billingProject = RawlsBillingProject(defaultBillingProjectName, CreationStatuses.Ready, Option(defaultBillingAccountName), None, googleProjectNumber = Option(defaultGoogleProjectNumber))
+      val v1Workspace = Workspace(billingProject.projectName.value, "v1", UUID.randomUUID().toString, "bucketName", None, DateTime.now, DateTime.now, "creator@example.com", Map.empty, false, WorkspaceVersions.V1, GoogleProjectId(billingProject.projectName.value), billingProject.googleProjectNumber, billingProject.billingAccount, None)
+      val v2Workspace = Workspace(billingProject.projectName.value, "v2", UUID.randomUUID().toString, "bucketName", None, DateTime.now, DateTime.now, "creator@example.com", Map.empty, false, WorkspaceVersions.V2, GoogleProjectId("differentId"), Option(GoogleProjectNumber("43")), billingProject.billingAccount, None)
+      val newBillingAccount = RawlsBillingAccountName("new-ba")
+
+      val failingGcsDAO = mock[GoogleServicesDAO](RETURNS_SMART_NULLS)
+      val failureMessage = "because I feel like it"
+      val exception = new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.Forbidden, failureMessage))
+      when(failingGcsDAO.updateGoogleProjectBillingAccount(any[GoogleProjectId], any[Option[RawlsBillingAccountName]]))
+        .thenReturn(Future.failed(exception))
+
+      runAndWait(rawlsBillingProjectQuery.create(billingProject))
+      runAndWait(rawlsBillingProjectQuery.load(billingProject.projectName)).getOrElse(fail("project not found"))
+        .invalidBillingAccount shouldBe false
+
+      runAndWait(workspaceQuery.createOrUpdate(v1Workspace))
+      runAndWait(workspaceQuery.createOrUpdate(v2Workspace))
+      runAndWait(rawlsBillingProjectQuery.updateBillingAccount(billingProject.projectName, Option(newBillingAccount)))
+
+      val actor = createWorkspaceBillingAccountMonitor(dataSource, failingGcsDAO)
+
+      eventually (timeout = timeout(Span(10, Seconds))) {
+        runAndWait(rawlsBillingProjectQuery.load(billingProject.projectName)).getOrElse(fail("project not found"))
+          .invalidBillingAccount shouldBe true
```

Comment:
In this scenario, we do not set the `billingAccountErrorMessage`s on the Workspaces, so I think we should add another assertion to collect the error messages on the Billing Project's workspaces and confirm that they're all blank.